lib/graphql/preload/instrument.rb

```diff
@@ -56,8 +65,10 @@ def instrument(_type, field)
         Promise.all(promises)
       end
 
-      private def preload_single_association(record, association)
-        GraphQL::Preload::Loader.for(record.class, association).load(record)
+      private def preload_single_association(record, association, scope_lambda, scope)
+        loader = GraphQL::Preload::Loader.for(record.class, association, scope_lambda)
```

Comment:
Did you use `scope_lambda` to generate a more unique loader key? Is there a way to avoid passing `scope_lambda` all the way down here? What about using `scope.to_sql` (or in Rails 5, `scope.cache_key`) instead?

Reply:
Yeah, we needed the loader key.  The problem we needed a loader key that wouldn't change from invocation to invocation.  I didn't think about using scope.to_sql, that could work.

Reply:
Sorry finally found some time to look at this.  You we're totally right.  I should be using scope.to_sql, which is working great.  I'm no longer passing `scope_lambda` all over the place.  Much cleaner.